src/IO/WriteBufferDecorator.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <IO/WriteBuffer.h>
+#include <IO/BufferWithOwnMemory.h>
+#include <utility>
+#include <memory>
+
+namespace DB
+{
+
+class WriteBuffer;
+
+/// WriteBuffer that works with another WriteBuffer.
```

Comment:
Comment is needed why this decorator is needed/what is does.
Also maybe we can use more specific name, but I don't know which one.

Reply:
> Also maybe we can use more specific name, but I don't know which one.

I chose `WriteBufferDecorator` because we decorate data and write it to underlying buffer. But maybe we can use `WriteBufferWithUnderlyingBuffer` or `WriteBufferWithNestedBuffer`, what do you think?